joblib/hashing.py

```diff
@@ -7,22 +7,17 @@
 # Copyright (c) 2009 Gael Varoquaux
 # License: BSD Style, 3 clauses.
 
-import warnings
-import pickle
 import hashlib
+import io
 import sys
 import types
-import struct
-from ._compat import _bytes_or_unicode
+import warnings
 
-import io
+from ._compat import _bytes_or_unicode
+from .externals import dill as pickle
```

Comment:
Not sure that's a great idea importing it as pickle, this could be rather confusing. Just stick with the 'dill' name.